### DISCUSSION TITLE
Use boolean to merge the taper

### DISCUSSION THREAD

**Discussion by YuleiDing** [gdsfactory/gdsfactory#486] [Q&A] [answered]
If I use the boolean to merge the taper and rectangle, the taper will become a rectangle, I don't why this happened? These are my codes.

import gdsfactory as gf
c=gf.Component("ding0628")
d=gf.Component("ding0629")
taper1=gf.components.taper(length=100, width1=8, width2=20)
rect1=c<<gf.components.rectangle(size=[20,8])
rect1.move([100,-4]);
t=d<<gf.geometry.boolean(taper1,rect1,operation="A+B")
d.plot()
d.show()

This is my result.
![20220628195427](https://user-images.githubusercontent.com/105275217/176172423-887de5d0-240f-4dfd-bf17-ab723de7cd79.png)

Before I update the gdsfactory, there isn't this question. Is this the mistake of boolean?

**Comment by joamatab** [comment, marked as answer]
https://github.com/gdsfactory/gdsfactory/issues/487
